pip/vcs/git.py

```diff
@@ -134,7 +134,7 @@ def obtain(self, dest):
             rev_options = [rev]
             rev_display = ' (to %s)' % rev
         else:
-            rev_options = ['origin/master']
+            rev_options = ['refs/remotes/origin/HEAD']
```

Comment:
It seems to me there's value in doing both (which is why I said "instead of _only_"): you can have the high-level test to check that the issue was fixed as reported. And you can have the lower-level, more focused test to check the function that you changed.

As far as making the lower-level test more sensible and less fragile, you'd probably know better than me what's appropriate. It seems like there'd be _something_ worth checking about that function's behavior that makes sense. Instead of mocking the subprocess call, is there something else you can do to trigger the necessary pre-condition (e.g. call a different method on the `Git` class)?